server/http_test.go

```diff
@@ -842,3 +949,593 @@ func TestGetSubscriptionsForChannel(t *testing.T) {
+
+			api.On("GetChannelMember", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(&model.ChannelMember{}, (*model.AppError)(nil))
+			api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(&model.Post{}, nil)
+
+			if tc.apiCalls != nil {
+				tc.apiCalls(api)
+			}
+
+			p.updateConfig(func(conf *config) {
+				conf.Secret = someSecret
+			})
+
+			p.SetAPI(api)
+			p.userStore = mockUserStore{}
+			p.instanceStore = p.getMockInstanceStoreKV(1)
+
+			w := httptest.NewRecorder()
+
+			request := httptest.NewRequest("DELETE",
+				"/api/v2/subscriptionTemplates/"+tc.templateID+"?instance_id="+testInstance1.GetID().String()+"&project_key=myproject",
+				nil)
+
+			if !tc.skipAuthorize {
+				request.Header.Set("Mattermost-User-Id", model.NewId())
+			}
+
+			p.ServeHTTP(&plugin.Context{}, w, request)
+			body, _ := ioutil.ReadAll(w.Result().Body)
+			t.Log(string(body))
+			assert.Equal(t, tc.expectedStatusCode, w.Result().StatusCode)
+		})
+	}
+}
+
+func TestEditSubscriptionTemplate(t *testing.T) {
+	count := 0
+	for name, tc := range map[string]struct {
+		subscriptionTemplate       string
+		expectedStatusCode int
+		skipAuthorize      bool
+		apiCalls           func(*plugintest.API)
+	}{
+		"Not Authorized": {
+			subscriptionTemplate:       "{}",
+			expectedStatusCode: http.StatusUnauthorized,
+			skipAuthorize:      true,
+		},
+		"Won't Decode": {
+			subscriptionTemplate:       "{woopsie",
+			expectedStatusCode: http.StatusBadRequest,
+		},
+		"Editing subscription template": {
+			subscriptionTemplate: `{
+				"instance_id": "jiraurl1",
+				"name": "some name",
+				"id": "aaaaaaaaaaaaaaaaaaaaaaaaab",
+				"filters": {
+				  "events": [
+					"jira:issue_created"
+				  ],
+				  "projects": [
+					"otherproject"
+				  ],
+				  "issue_types": [
+					"10001"
+				  ],
+				  "fields": []
+				}
+			  }`,
+			expectedStatusCode: http.StatusOK,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{
+				{
+					ID: "aaaaaaaaaaaaaaaaaaaaaaaaab",
+					Filters: SubscriptionFilters{
+						Events:     NewStringSet("jira:issue_created"),
+						Projects:   NewStringSet("otherproject"),
+						IssueTypes: NewStringSet("10001"),
+					},
+				},
+			},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID: "aaaaaaaaaaaaaaaaaaaaaaaaab",
+							Filters: SubscriptionFilters{
+								Events:     NewStringSet("jira:issue_created"),
+								Projects:   NewStringSet("myproject"),
+								IssueTypes: NewStringSet("10001"),
+							},
+						},
+					}), t),
+		},
+		"Editing subscription template, no name provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "", "id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaac", "filters": {"events": ["jira:issue_created"], "projects": ["otherproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode: http.StatusInternalServerError,
```

Comment:
just a query. should the status code be BadRequest or InternalServerError?

Reply:
Here in this, there is a common function checking the fields for both subscriptions and subscription templates.This error is coming from there that's why I am not changing the status code as it is already existing code